system/jlib/jatomic.hpp

```diff
@@ -197,10 +197,13 @@ typedef volatile long atomic_t;
 
 #elif defined(__GNUC__)
 
-typedef struct { volatile int counter; } atomic_t;
 #define ATOMIC_INIT(i)          { (i) }
 #define atomic_read(v)          ((v)->counter)
+
```

Comment:
This change doesn't match the one in eclrtl_imp.hpp.  That added

#ifndef atomic_type_defined

All of this code should be deleted anyway, but I have created a separate jira for that.